code/datums/sleep_advancement.dm

```diff
@@ -244,15 +244,16 @@
 		dat += "<br><a [can_buy ? "" : "class='linkOff'"] href='byond://?src=[REF(src)];task=buy_special'>Dream something <b>special</b></a> - \Roman[get_special_cost()]"
 		dat += "<br>Specials can have negative or positive effects"
 	dat += "<div class='footer'>"
-	dat += "<br><br><center>Your points will be retained<br><a href='byond://?src=[REF(src)];task=continue'>Continue</a></center>"
+	dat += "<br><br><center><span style='color:#968160;'>Your points will be retained<br></span><a href='byond://?src=[REF(src)];task=continue' style='color:#503636; text-decoration:none; background-color:black;'>Continue</a></center>"
 	dat += {"
 			</body>
 		</html>
 	"}
-	var/datum/browser/popup = new(user, "dreams", "<center>Dreams</center>", 350, 450)
-	popup.set_window_options(can_close = FALSE)
-	popup.set_content(dat.Join())
-	popup.open(FALSE)
+
+	user << browse(
```

Comment:
???

Reply:
The problem is that set_window_options don't works at all even if i set everything false, you can close or do whatever and the close_ui() keeps firing all the time even outside of sleep when i tested it with logs.

The only thing i did was copy from the adv class pop up that uses browse because that actually works and prevents people from closing the popup.

But i am trying a few other stuff

Reply:
if close_ui() always runs then just move the confirm behaviour to that and make the X and confirm do the same thing